src/SimulationService.jl

```diff
@@ -1,77 +1,472 @@
+    # If JobID is not found in the TDS, we will create it
+    obj = isnothing(obj) ? Dict() : Dict(obj)
+
+    body = JSON3.write(merge(Dict(obj), Dict(kw)))
+    HTTP.put(url, JSON_HEADER; body=body)
+end
+
+
+function upload_results!(o::OperationRequest)
+    isnothing(o.results) && error("No results.  Run `solve!(o)` first.")
+
+    # DataFrame result saved as CSV.  Everything else saved as JSON.
+    o.results_to_upload = if o.results isa DataFrame
+        io = IOBuffer()
+        CSV.write(io, o.results)
+        body = String(take!(io))
+        filename = "result.csv"
+        header = ["Content-Type" => "text/csv"]
+    else
+        body = JSON3.write(o.results)
+        filename = "result.json"
+        header = JSON_HEADER
+    end
+
+    if !SIMSERVICE_ENABLE_TDS
+        return no_tds(:upload_results!; filename, header, bodysummary=repr(summary(body)))
+    end
+
+    upload_url = "$SIMSERVICE_TDS_URL/simulations/sciml-$(o.job_id)/upload-url?filename=$filename)"
+    url = get_json(upload_url).url
+    HTTP.put(url, header; body=body)
+end
+
+#-----------------------------------------------------------------------------# solve!
+function solve!(o::OperationRequest)
+    try
+        update_job_status!(o; status="running", start_time=time())
+        operation = o.operation_type(o)
+        callback = get_callback(o)
+        o.results = solve(operation; callback)
+        upload_results!(o)
+        update_job_status!(o; status="complete", complete_time=time())
+        return o.results_to_upload
+    catch ex
+        update_job_status!(o; status="error", error=string(ex))
+    end
+end
+
+#-----------------------------------------------------------------------------# POST /{operation}
+# For debugging.  When a job fails, you can check out last_operation[] and last_job[].
+last_operation = Ref{OperationRequest}()
+last_job = Ref{JobSchedulers.Job}()
+
+# TODO: add try-catch back in?  It's useful for debugging to leave it out.
+function operation(req::HTTP.Request, operation_name::String)
+    # try
+        o = OperationRequest(req, operation_name)
+        @info "Scheduling Job: $(o.job_id)"
+        job = JobSchedulers.Job(@task(solve!(o)))
+        job.id = jobhash(o.job_id)
+        JobSchedulers.submit!(job)
+
+        last_operation[] = o    # For debugging
+        last_job[] = job        # For debugging
+
+        body = JSON3.write((; simulation_id = o.job_id))
+        return HTTP.Response(201, ["Content-Type" => "application/json; charset=utf-8"], body; request=req)
+    # catch ex
+    #     return HTTP.Response(500, ["Content-Type" => "application/json; charset=utf-8"], JSON3.write((; error=string(ex))))
+    # end
+end
+
+
+#-----------------------------------------------------------------------------# simulate
+struct Simulate <: Operation
+    sys::ODESystem
+    timespan::Tuple{Float64, Float64}
+end
+
+Simulate(o::OperationRequest) = Simulate(ode_system_from_amr(o.model), o.timespan)
```

Comment:
out of curiosity, where it is known that OperationRequest has o.model and o.timespan?

Reply:
I'm not sure I follow.  Those are fields of the struct.

Reply:
Check `examples/` to see what the incoming requests look like.  E.g.

```json
{
  "model_config_id": "f20ea672-bfa4-422b-884f-0a4c09356fd0",
  "timespan": {"start": 0, "end": 90}
}
```

We then reach out to the TDS to get the the AMR (`model`) from the given `model_config_id`.